executor/joiner.go

```diff
@@ -575,12 +646,84 @@ func (j *leftOuterSemiJoiner) Clone() joiner {
 	return &leftOuterSemiJoiner{baseJoiner: j.baseJoiner.Clone()}
 }
 
+type nullAwareAntiLeftOuterSemiJoiner struct {
+	baseJoiner
```

Comment:
Can we put the NAAJType as a member variable of the struct?

Reply:
NAAJType is not a fixed joiner parameter, but the runtime status according to the left and right rows characteristic
<img width="707" alt="image" src="https://user-images.githubusercontent.com/15825830/189065647-edb87307-1f22-40eb-9433-046df7778595.png">

Reply:
now it is only used by AntiLeftOuterSemiJoiner to fill the specified result according to runtime NAAJType.
<img width="862" alt="image" src="https://user-images.githubusercontent.com/15825830/189066290-40685b63-16a0-4933-9fa1-6652aa9b35f6.png">

Reply:
On only the OnMisMatch, we will fill it will `1`

Reply:
Looks like null-aware join is so different with the normal join, in this pr introduce too many `if isNA` related code, even the top interface `tryMatchInners/onMatch` are polluted. I wonder that comparing to the current implementation, if it can be more clear by adding a new NAHashJoinExec instead?  @XuHuaiyu what do you think?

Reply:
Yeah, i'm also thinking whether we could split to two join exec that extract the comment part as an interface and baseHashJoinExec.

Reply:
That seems reasonable, the code size will expand dramatically in this pr (some copy work is unavoidable even if extracting an interface for baseHashJoinExec), we can put the clear work in the next pr?